src/Engine/UnitActionLifecycles/UnitProduceLifecycle.java

```diff
@@ -150,7 +150,8 @@ public void performEvent(MapMaster gameMap)
       if( null != myNewUnit )
       {
         myCommander.army.money -= cost;
-        builder.materials -= 1;
+        if( builder.model.maxMaterials >= 0 )
+          builder.materials -= 1;
```

Comment:
Why not use `hasMaterials()` here as well?

Reply:
This commit adds support for units to have unlimited materials via a max value of -1.
I don't want to decrement materials past that sentinel value.